README.md

```diff
@@ -28,27 +28,32 @@ prime-order group from a non-prime-order Edwards curve.  This provides the
 speed and safety benefits of Edwards curve arithmetic, without the pitfalls of
 cofactor-related abstraction mismatches.
 
-# Documentation
-
-The `curve25519-dalek` public API and the backends are documented [here][docs-external].
-
-In addition, the unstable internal implementation can be generated locally as below.
-
-```sh
-make doc-internal
-```
-
 # Use
 
 To import `curve25519-dalek`, add the following to the dependencies section of
 your project's `Cargo.toml`:
 ```toml
-curve25519-dalek = "4.0.0-pre.2"
+curve25519-dalek = "4"
 ```
 
+## Feature Flags
+
+* `std` [default] - Enables `alloc`
+and `rand_core/std`, which
+[slightly expands](https://github.com/rust-random/rand/tree/7aa25d577e2df84a5156f824077bb7f6bdf28d97/rand_core#crate-features)
+the number of things that impl `RngCore`.
+* `alloc` [default] - Implied by `std`. This enables Edwards and Ristretto multiscalar multiplication, batch scalar inversion, and batch Ristretto double-and-compress.
+* `rand_core` - Enables the functions `Scalar::random` and `RistrettoPoint::random`. This is an optional dependency whose version is not subject to SemVer. See [below](#public-api-semver-exemptions) for more details.
+* `digest` - Enables the functions `RistrettoPoint::{from_hash, hash_from_bytes}` and `Scalar::{from_hash, hash_from_bytes}`. This is an optional dependency whose version is not subject to SemVer. See [below](#public-api-semver-exemptions) for more details.
+* `serde` - Enables `serde` serialization/deserialization for all the point and scalar types.
+* `simd_backend` - See [backends](#backends). Requires nightly.
+* `fiat_backend` - See [backends](#backends).
+
+To disable the default features when using `curve25519-dalek` as a dependency, add `default-features = false` to the dependency in your `Cargo.toml`. To disable it when running `cargo`, add the `--no-default-features` CLI flag.
+
 ## Major Version API Changes
 
-See `CHANGELOG.md` for more details.
+See [`CHANGELOG.md`](CHANGELOG.md) for more details.
 
 ### 2.x
```

Comment:
I would move these to CHANGELOG.md and mark breaking public API changes some way e.g. `Changed:` there

Reply:
I kinda of agree that it feels cluttered here, but CHANGELOG.md has a list of _all_ the versions. So to keep it consistent, the API-breaking changes would have to go just in the x.0.0 headers. But that makes it hard to skim. Did you have something in mind?

Reply:
Ctrl-F "Breaking Changes" <Enter> <Enter> ?
Ideally they probably mayhaps be in Release notes in GitHub and then link to release notes - which in turn get generated from CHANGELOG.md automatically upon release - that could be linked instead ?
Could also link by to CHANGELOG.md with `#` anchor points ?

There isn't too many with 2.x anymore and the ones at 2.x are unmaintained anyways me thinks ?
https://crates.io/crates/curve25519-dalek/reverse_dependencies

Reply:
I split the difference a bit here. I left the 4.x breaking changes in the README so people can see it at a glance. The rest is now in the CHANGELOG under "Breaking changes"